prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,341 @@
+
+    /**
+     * Create Email Schedule
+     */
+    @FunctionName("createEmailSchedule")
+    @StorageAccount("AzureWebJobsStorage")
+    fun createEmailSchedule(
+        @HttpTrigger(
+            name = "createEmailSchedule",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-schedule"
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        var user:String? = validateUser( request, context.logger );
+        var ret = request.createResponseBuilder(HttpStatus.UNAUTHORIZED);
+
+        if( user !== null ){
+            val id = WorkflowEngine.databaseAccess.insertEmailSchedule(request.body, user );
+            ret.status(HttpStatus.CREATED)
+            ret.body( "created id $id")
+        }
+        return ret.build();
+    }
+
+    /**
+     * Delete (inactivate) Email Schedule
+     */
+    @FunctionName("deleteEmailSchedule")
+    @StorageAccount("AzureWebJobsStorage")
+    fun deleteEmailSchedule(
+        @HttpTrigger(
+            name = "deleteEmailSchedule",
+            methods = [HttpMethod.DELETE],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-schedule/{scheduleId}"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("scheduleId") scheduleId: Int,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        var user:String? = validateUser( request, context.logger );
+        var ret = request.createResponseBuilder(HttpStatus.UNAUTHORIZED);
+
+        if( user !== null ){
+            val id = WorkflowEngine.databaseAccess.deleteEmailSchedule(scheduleId);
+            ret.status(HttpStatus.CREATED)
+            ret.body( "deactivated id $id")
+        }
+        return ret.build();
+    }
+
+    /**
+     * Timer Trigger to fire when processing schedules
+     *   (fires every 5 minutes)
+     */
+    @FunctionName("emailScheduleEngine")
+    @StorageAccount("AzureWebJobsStorage")
+    fun run(
+        @TimerTrigger( name = "emailScheduleEngine", schedule = SCHEDULE ) timerInfo : String,
+        context: ExecutionContext
+    ){
+        val logger : Logger = context.logger;
+        val mapper = ObjectMapper().registerModule(KotlinModule());
+
+        // get the schedules to fire
+        val schedulesToFire : List<EmailSchedule> = WorkflowEngine.databaseAccess.fetchEmailSchedules()
+            .map{ mapper.readValue<EmailSchedule>( it ) }
+            .filter{ shouldFire( it ) }
+
+        schedulesToFire.forEach { schedule -> 
+            val orgs: Iterable<String> = getOrgs( schedule );
+
+            logger.info( "processing ${schedule.template}" )
+
+            // get the orgs to fire for
+            orgs.forEach{ org ->
+                    val emails: List<String> = if ( schedule.emails.size > 0) schedule.emails else getEmails(org, logger)
+                    logger.info( "processing ${org}" )
```

Comment:
I would make this a bit more explanatory in terms of _what_ we are processing. Otherwise we may be making our lives harder than needed when we need to trawl through logs to figure out what's happening.

Reply:
OK